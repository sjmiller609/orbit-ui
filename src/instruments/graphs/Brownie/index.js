'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { Row, B, FormLabel } from 'instruments'
import { colorScale } from 'helpers/colors'

const Brownie = ({ slices, part, total, title, convert, className }) => {
  const colors = colorScale(slices.length)

  return (
    <div className={classnames(s.chart, className)}>
      {title && <FormLabel>{title}</FormLabel>}
      <Row justify="flex-start">
        <Row className={s.bar}>
          {slices.map((m, i) => {
            const r = Math.floor(m.value / total * 100) / 100
            return (
              <div
                key={i}
                className={s.slice}
                style={{
                  color: colors[i].a,
                  background: `linear-gradient(45deg, ${colors[i].a}, ${
                    colors[i].z
                  } 160%)`,
                  width: `${r * 100}%`,
                }}>
                <label>
                  {m.name}
                  <B>
                    {m.value}AU ({convert(m.value)})
                  </B>
                </label>
              </div>
            )
          })}
          {total - part > 0 && (
            <div
              className={classnames(s.slice, s.remaining)}
              style={{
                width: `${Math.floor((total - part) / total) * 100}%`,
              }}>
              <label>
                Remaining
                <B>{convert(total - part)}</B>
              </label>
            </div>
          )}
        </Row>
      </Row>
    </div>
  )
}

Brownie.propTypes = {
  slices: PropTypes.array,
  className: PropTypes.string,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  convert: PropTypes.func,
  total: PropTypes.number,
  part: PropTypes.number,
}

Brownie.defaultProps = {
  convert: v => v,
}

export default Brownie
