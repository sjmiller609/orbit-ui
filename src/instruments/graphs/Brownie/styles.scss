@import 'src/styles/mixins.scss';

.bar {
  width: 100%;
  padding: 0.5rem;
  border-radius: 6px;
  border: 2px solid $lightGray;
  margin: 0.8rem 0;
  justify-content: flex-start;
}

.slice {
  $height: 2rem;
  height: $height;
  display: block;
  position: relative;
  min-width: 6px;
  &:not(:last-child) {
    border-right: 3px solid $white;
  }
  &:first-child {
    border-top-left-radius: 4px;
    border-bottom-left-radius: 4px;
  }
  &:last-child {
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
  }
  &:hover label {
    opacity: 1;
    z-index: 1000;
  }
  label {
    z-index: -1000;
    opacity: 0;
    display: inline-block;
    white-space: nowrap;
    text-align: center;
    position: absolute;
    bottom: 0.8rem;

    text-transform: capitalize;
    @include font($f400);
    line-height: 1.4em;
    background: $white;
    border: 1px solid;
    font-size: 0.88em;
    position: relative;
    @include shadow;
    padding: 0.8em 1em;
    border-radius: 4px;

    left: 50%;
    transform: translate(-50%, -100%);
    &:before {
      content: '';
      display: block;
      background: $white;
      height: 1rem;
      width: 1rem;
      transform: translateX(-50%) rotate(45deg);
      position: absolute;
      border-radius: 2px;
      border-right: 1px solid;
      border-bottom: 1px solid;
      bottom: -0.6rem;
      left: 50%;
      z-index: 2000;
    }
    b {
      display: block;
    }
  }
  &.remaining {
    background: $lightGray;
    color: $beige;
  }
}
