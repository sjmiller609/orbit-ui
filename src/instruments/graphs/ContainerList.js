import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import s from './styles.scss'

class ContainerList extends React.Component {
  state = {
    data: [],
  }

  componentDidUpdate(prevProps) {
    if (prevProps != this.props) {
      this.formatData(this.props)
    }
  }

  componentWillMount() {
    this.formatData(this.props)
  }

  formatData = props => {
    const { metric } = props
    this.setState({ data: metric })
  }

  formatBool = value => {
    const isTrue = value === '0'
    if (isTrue)
      return (
        <span className={classnames(s.healthLabel, s.healthy)}>Healthy</span>
      )
    return (
      <span className={classnames(s.healthLabel, s.unhealthy)}>Unhealthy</span>
    )
  }

  render() {
    const { data } = this.state

    if (data.length > 0) {
      data.sort((a, b) => {
        let textA = a.metric.container.toUpperCase()
        let textB = b.metric.container.toUpperCase()
        return textA < textB ? -1 : textA > textB ? 1 : 0
      })
      return (
        <div className={s.containerListContainer}>
          <div className={s.containerListHeader}>
            <div>Pod</div>
            <div style={{ flex: '0 0 45%' }}>Container</div>
            <div style={{ textAlign: 'center', flex: '0 0 10%' }}>Status</div>
          </div>
          <div className={s.containerListItems}>
            {data.map(
              (d, i) =>
                d.metric.container && (
                  <div
                    key={`${d.metric.container}${i}`}
                    className={s.containerListItem}>
                    <div>{d.metric.pod}</div>
                    <div style={{ flex: '0 0 45%' }}>{d.metric.container}</div>
                    <div style={{ textAlign: 'center', flex: '0 0 10%' }}>
                      {this.formatBool(d.value[1])}
                    </div>
                  </div>
                )
            )}
          </div>
        </div>
      )
    }

    return null
  }
}

ContainerList.propTypes = {
  metric: PropTypes.array,
  label: PropTypes.string,
}

export default ContainerList
