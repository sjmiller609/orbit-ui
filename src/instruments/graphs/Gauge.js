import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { ResponsivePieCanvas } from '@nivo/pie'
import { P } from 'instruments'

import s from './styles.scss'

const levelLabels = [
  {
    id: 'danger',
    value: 25,
    color: '#f4362c',
  },
  {
    id: 'warn',
    value: 25,
    color: '#ff9a09',
  },
  {
    id: 'ok',
    value: 50,
    color: '#00ac6b',
  },
]

class Gauge extends React.Component {
  state = {
    value: 0,
  }

  componentDidUpdate(prevProps) {
    if (prevProps != this.props) {
      this.formatData(this.props)
    }
  }

  componentWillMount() {
    this.formatData(this.props)
  }

  formatData = props => {
    const { metric } = props
    if (metric != undefined && metric.length > 0) {
      this.setState({ value: Math.round(Number(metric[1])) })
    }
  }

  formatColor = value => {
    if (value <= 50) {
      return '#00ac6b'
    }

    if (value > 50 && value <= 75) {
      return '#ff9a09'
    }

    return '#f4362c'
  }

  render() {
    const { value } = this.state

    return (
      <div className={s.gaugeContainer}>
        <div className={s.gaugeElement}>
          <ResponsivePieCanvas
            data={levelLabels}
            width={300}
            height={200}
            pixelRatio={2}
            startAngle={90}
            endAngle={-90}
            sortByValue={false}
            innerRadius={0.95}
            enableRadialLabels={false}
            enableSlicesLabels={false}
            isInteractive={false}
            colors={d => d.color}
          />
        </div>
        <div className={classnames(s.gaugeElement, s.guagePercent)}>
          <ResponsivePieCanvas
            data={[
              {
                id: 'reserve',
                value: 100 - value,
                color: '#f2f1ee',
              },
              {
                id: 'used',
                value: value,
                color: this.formatColor(value),
              },
            ]}
            width={300}
            height={200}
            pixelRatio={2}
            startAngle={90}
            endAngle={-90}
            sortByValue={false}
            innerRadius={0.75}
            enableRadialLabels={false}
            enableSlicesLabels={false}
            isInteractive={false}
            colors={d => d.color}
          />
        </div>
        <div className={s.gaugeLabel}>
          <P>{isNaN(value) ? 0 : value}%</P>
        </div>
      </div>
    )
  }
}

Gauge.propTypes = {
  metric: PropTypes.array,
}

export default Gauge
