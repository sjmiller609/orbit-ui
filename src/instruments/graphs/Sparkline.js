import React from 'react'
import PropTypes from 'prop-types'
import { Sparklines, SparklinesCurve } from 'react-sparklines'
import { P } from 'instruments'

import s from './styles.scss'

class Sparkline extends React.Component {
  state = {
    data: [],
    current: 0,
  }

  componentDidUpdate(prevProps) {
    if (prevProps != this.props) {
      this.formatData(this.props)
    }
  }

  componentWillMount() {
    this.formatData(this.props)
  }

  formatData = props => {
    const { metric } = props

    const data = []
    metric.map(v => data.push(v[1]))

    const length = data.length - 1
    this.setState({ data, current: data[length] })
  }

  render() {
    const { data, current } = this.state

    return (
      <div className={s.sparklineContainer}>
        <div className={s.sparklineLabel}>
          <P>{`${isNaN(Math.round(current)) ? 0 : Math.round(current)}`}</P>
        </div>
        <div className={s.sparklineElement}>
          <Sparklines data={data} limit={50} width={400} height={50} margin={0}>
            <SparklinesCurve color="#0CBEDB" />
          </Sparklines>
        </div>
      </div>
    )
  }
}

Sparkline.propTypes = {
  metric: PropTypes.array,
  label: PropTypes.string,
}

export default Sparkline
