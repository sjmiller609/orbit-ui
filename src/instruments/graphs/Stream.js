import React from 'react'
import PropTypes from 'prop-types'
import { ResponsiveStream } from '@nivo/stream'
import { P } from 'instruments'

import s from './styles.scss'

class Stream extends React.Component {
  state = {
    data: [],
    length: 0,
  }

  componentDidUpdate(prevProps) {
    if (prevProps != this.props) {
      this.formatData(this.props)
    }
  }

  componentWillMount() {
    this.formatData(this.props)
  }

  formatData = props => {
    const { metric } = props
    const data = []
    let sum = 0

    if (metric[0] === undefined) return

    const length =
      metric[0].result.length != 0 && metric[0].result[0].values != undefined
        ? metric[0].result[0].values.length
        : 30

    try {
      Array.apply(null, Array(length)).map((v, i) => {
        const slice = {
          'Queued Tasks':
            metric[0].result.length != 0 &&
            metric[0].result[0].values[i] != undefined
              ? Math.round(Number(metric[0].result[0].values[i][1]))
              : 0,
          'Running Tasks':
            metric[1].result.length != 0 &&
            metric[1].result[0].values[i] != undefined
              ? Math.round(Number(metric[1].result[0].values[i][1]))
              : 0,
          'Successful Tasks':
            metric[2].result.length != 0 &&
            metric[2].result[0].values.reverse()[length - i] != undefined
              ? Math.round(
                  Number(metric[2].result[0].values.reverse()[length - i][1])
                )
              : 0,
          'Failed Tasks':
            metric[3].result.length != 0 &&
            metric[3].result[0].values.reverse()[length - i] != undefined
              ? Math.round(
                  Number(metric[3].result[0].values.reverse()[length - i][1])
                )
              : 0,
        }
        data.push(slice)
        sum += Object.values(slice).reduce((a, b) => a + b)
      })
    } catch (ignore) {
      // Ignore this error. This can happen before data is completely ready in prometheus.
    }

    if (sum > 0) this.setState({ length, data })
    else this.setState({ length: 0, data: [] })
  }

  render() {
    const { length, data } = this.state
    const colors = ['#C8C3BD', '#07d65b', '#049a59', '#FF523A'].reverse()

    return (
      <div className={s.streamContainer}>
        {length === 0 && (
          <div className={s.streamNoData}>
            <P>No task data</P>
          </div>
        )}
        <div className={s.streamElement} style={{ height: 200 }}>
          <ResponsiveStream
            data={data}
            keys={[
              'Queued Tasks',
              'Running Tasks',
              'Successful Tasks',
              'Failed Tasks',
            ].reverse()}
            axisBottom={null}
            offsetType="silhouette"
            colors={c => colors[c.index]}
            enableGridX={false}
            margin={{ top: 0, right: 150, bottom: 30, left: 40 }}
            fillOpacity={0.85}
            animate={false}
            legends={[
              {
                anchor: 'top-right',
                direction: 'column',
                justify: false,
                translateX: 65,
                translateY: 0,
                itemsSpacing: 0,
                itemDirection: 'left-to-right',
                itemWidth: 50,
                itemHeight: 20,
                itemOpacity: 0.75,
                symbolSize: 5,
                symbolShape: 'square',
              },
            ]}
          />
        </div>
      </div>
    )
  }
}

Stream.propTypes = {
  metric: PropTypes.array,
  label: PropTypes.string,
}

export default Stream
