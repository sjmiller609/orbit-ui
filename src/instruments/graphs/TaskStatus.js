import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import s from './styles.scss'

class TaskStatus extends React.Component {
  state = {
    data: [],
  }

  componentDidUpdate(prevProps) {
    if (prevProps != this.props) {
      this.formatData(this.props)
    }
  }

  componentWillMount() {
    this.formatData(this.props)
  }

  formatData = props => {
    const { metric } = props
    this.setState({ data: metric })
  }

  formatBool = value => {
    const isTrue = value === '0'
    if (isTrue)
      return (
        <span className={classnames(s.healthLabel, s.healthy)}>Healthy</span>
      )
    return (
      <span className={classnames(s.healthLabel, s.unhealthy)}>Unhealthy</span>
    )
  }

  render() {
    const { data } = this.state

    if (data.length > 0)
      return (
        <div className={s.containerListContainer}>
          <div className={s.containerListHeader}>
            <div>Pod</div>
            <div style={{ flex: '0 0 45%' }}>Container</div>
            <div style={{ textAlign: 'center', flex: '0 0 10%' }}>Status</div>
          </div>
          <div className={s.containerListItems}>
            {data.map((d, i) => (
              <div
                key={`${d.metric.container}${i}`}
                className={s.containerListItem}>
                <div>{d.metric.pod}</div>
                <div style={{ flex: '0 0 45%' }}>{d.metric.container}</div>
                <div style={{ textAlign: 'center', flex: '0 0 10%' }}>
                  {this.formatBool(d.value[1])}
                </div>
              </div>
            ))}
          </div>
        </div>
      )

    return null
  }
}

TaskStatus.propTypes = {
  metric: PropTypes.array,
  label: PropTypes.string,
}

export default TaskStatus
