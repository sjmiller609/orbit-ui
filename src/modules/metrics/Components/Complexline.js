import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { ResponsiveLine } from '@nivo/line'

import s from './styles.scss'

class Complexline extends React.Component {
  state = {
    data: [],
    emptyData: [],
  }

  componentDidUpdate(prevProps) {
    if (prevProps != this.props) {
      this.formatData(this.props)
    }
  }

  componentWillMount() {
    this.formatData(this.props)
  }

  formatValue = amount => {
    const { label } = this.props

    switch (label) {
      case 'GB':
        amount = (Math.round(amount) / 1073741824).toFixed(1)
        break
      case 'kB/s':
        amount = (Math.round(amount) / 1024).toFixed(0)
        break
      case '%':
        amount = amount.toFixed(0)
        break
      case 'ops':
        amount = amount.toFixed(2)
        break
      default:
        amount
    }

    return amount
  }

  formatData = props => {
    const { metric } = props
    const data = []
    const emptyData = []

    // Format the data OR generate placeholder for
    // data returned as 0 (so a graph will show)
    if (metric.length > 0) {
      metric.map((m, i) => {
        const id = m.metric.operator
          ? m.metric.operator
          : m.metric.pod_name.split(/-/g)[3] +
            '-' +
            m.metric.pod_name.split(/-/g)[4]
        return data.push({
          id: `${id}-${i}`,
          data: m.values.map(v => {
            return {
              x: moment.unix(v[0]).format('YYYY-MM-DD HH:mm:ss'),
              y: Number(v[1]),
            }
          }),
        })
      })
    } else {
      emptyData.push({
        id: 'No data',
        data: Array.apply(null, Array(10)).map((v, i) => {
          return {
            x: moment()
              .subtract(i, 'minutes')
              .format('YYYY-MM-DD HH:mm:ss'),
            y: Number(`0.000${i}`),
          }
        }),
      })
    }

    this.setState({ data, emptyData })
  }

  render() {
    const { data, emptyData } = this.state
    const { label, step } = this.props

    return (
      <div className={s.complexlineContainer}>
        <div className={s.complexlineElement} style={{ height: 300 }}>
          <ResponsiveLine
            data={data.length != 0 ? data : emptyData}
            margin={{ top: 20, right: 120, bottom: 80, left: 60 }}
            pixelRatio={2}
            curve="monotoneX"
            lineWidth={1}
            enableArea={true}
            isInteractive={data.length != 0}
            enablePoints={false}
            enableGridX={false}
            xScale={{
              type: 'time',
              format: '%Y-%m-%d %H:%M:%S',
            }}
            xFormat="time:%Hh %Mm $Ss"
            axisBottom={{
              format: () => null,
              tickSize: 0,
              tickValues: `every ${step} seconds`,
            }}
            yScale={{
              type: 'linear',
              stacked: false,
              min: 0,
              max: data.length != 0 ? 'auto' : 1,
            }}
            axisLeft={{
              orient: 'right',
              tickSize: 5,
              tickPadding: 5,
              tickRotation: 0,
              tickValues: 4,
              format: v =>
                `${Number(this.formatValue(v))}${
                  label != '%' ? ' ' : ''
                }${label}`,
            }}
            yFormat={v =>
              `${Number(this.formatValue(v))}${label != '%' ? ' ' : ''}${label}`
            }
            useMesh={true}
            enableSlices={data.length != 0 ? 'x' : false}
            legends={[
              {
                anchor: 'top-right',
                direction: 'column',
                justify: false,
                translateX: 65,
                translateY: 0,
                itemsSpacing: 0,
                itemDirection: 'left-to-right',
                itemWidth: 50,
                itemHeight: 20,
                itemOpacity: 0.75,
                symbolSize: 5,
                symbolShape: 'square',
              },
            ]}
            sliceTooltip={({ slice }) => {
              return (
                <div className={s.tooltipContainer}>
                  {slice.points.map(point => (
                    <div key={point.id} className={s.tooltip}>
                      <strong style={{ color: point.serieColor }}>
                        {point.serieId}
                      </strong>
                      {point.data.yFormatted}
                    </div>
                  ))}
                </div>
              )
            }}
          />
        </div>
      </div>
    )
  }
}

Complexline.propTypes = {
  metric: PropTypes.array,
  label: PropTypes.string,
  step: PropTypes.number,
}

export default Complexline
