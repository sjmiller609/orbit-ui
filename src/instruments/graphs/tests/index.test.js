import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Brownie from '../Brownie'
import Complexline from '../Complexline'
import ContainerList from '../ContainerList'
import Gauge from '../Gauge'
import Sparkline from '../Sparkline'
import Stream from '../Stream'
import TaskStatus from '../TaskStatus'

import { ResponsiveLine } from '@nivo/line'

const convert = jest.fn()

const brownieProps = {
  slices: ['1', '2', '3'],
  className: 'testClassString',
  title: 'Test',
  convert,
  total: 400,
  part: 100,
}

const mockMetric = {
  metric: [],
  label: 'AU',
}

const labels = ['GB', 'kB/s', '%', 'ops']

describe('Graphs', () => {
  let brownie
  let complexLine
  let containerList
  let gauge
  let sparkLine
  let stream
  let taskStatus

  beforeEach(() => {
    brownie = shallow(<Brownie {...brownieProps} />)
    complexLine = renderer.create(<Complexline {...mockMetric} />)
    containerList = shallow(<ContainerList {...mockMetric} />)
    gauge = shallow(<Gauge />)
    sparkLine = shallow(<Sparkline {...mockMetric} />)
    stream = shallow(<Stream {...mockMetric} />)
    taskStatus = shallow(<TaskStatus {...mockMetric} />)
  })

  it('should be defined', () => {
    expect(Brownie).toBeDefined()
    expect(Complexline).toBeDefined()
    expect(ContainerList).toBeDefined()
    expect(Gauge).toBeDefined()
    expect(Sparkline).toBeDefined()
    expect(Stream).toBeDefined()
    expect(TaskStatus).toBeDefined()
  })

  it('should render correctly', () => {
    expect(brownie).toMatchSnapshot()
    expect(complexLine).toMatchSnapshot()
    expect(containerList).toMatchSnapshot()
    expect(gauge).toMatchSnapshot()
    expect(sparkLine).toMatchSnapshot()
    expect(stream).toMatchSnapshot()
    expect(taskStatus).toMatchSnapshot()
  })

  it('graph imports should be defined', () => {
    expect(ResponsiveLine).toBeDefined()
  })

  it(`complexline should format labels correctly`, () => {
    labels.map(label => {
      const newComplexLine = renderer.create(<Complexline label={label} />)
      const inst = newComplexLine.getInstance()
      expect(inst.formatValue(123)).toMatchSnapshot()
    })
  })

  it(`complexline should format data correctly`, () => {
    labels.map(label => {
      const newComplexLine = renderer.create(<Complexline label={label} />)
      const inst = newComplexLine.getInstance()
      expect(inst.formatData([])).toMatchSnapshot()
    })
  })
})
