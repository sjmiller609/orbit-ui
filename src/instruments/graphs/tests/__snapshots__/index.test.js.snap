// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graphs complexline should format data correctly 1`] = `undefined`;

exports[`Graphs complexline should format data correctly 2`] = `undefined`;

exports[`Graphs complexline should format data correctly 3`] = `undefined`;

exports[`Graphs complexline should format data correctly 4`] = `undefined`;

exports[`Graphs complexline should format labels correctly 1`] = `"0.0"`;

exports[`Graphs complexline should format labels correctly 2`] = `"0"`;

exports[`Graphs complexline should format labels correctly 3`] = `"123"`;

exports[`Graphs complexline should format labels correctly 4`] = `"123.00"`;

exports[`Graphs should render correctly 1`] = `
<div
  className="chart testClassString"
>
  <FormLabel>
    Test
  </FormLabel>
  <Row
    justify="flex-start"
  >
    <Row
      className="bar"
    >
      <div
        className="slice"
        key="0"
        style={
          Object {
            "background": "linear-gradient(45deg, #00B35D, #0CBEDB 160%)",
            "color": "#00B35D",
            "width": "NaN%",
          }
        }
      >
        <label>
          <B>
            AU (
            )
          </B>
        </label>
      </div>
      <div
        className="slice"
        key="1"
        style={
          Object {
            "background": "linear-gradient(45deg, #0CBEDB, #302c43 160%)",
            "color": "#0CBEDB",
            "width": "NaN%",
          }
        }
      >
        <label>
          <B>
            AU (
            )
          </B>
        </label>
      </div>
      <div
        className="slice"
        key="2"
        style={
          Object {
            "background": "linear-gradient(45deg, #ff9a09, #f4362c 160%)",
            "color": "#ff9a09",
            "width": "NaN%",
          }
        }
      >
        <label>
          <B>
            AU (
            )
          </B>
        </label>
      </div>
      <div
        className="slice remaining"
        style={
          Object {
            "width": "75%",
          }
        }
      >
        <label>
          Remaining
          <B />
        </label>
      </div>
    </Row>
  </Row>
</div>
`;

exports[`Graphs should render correctly 2`] = `
<div
  className="complexlineContainer"
>
  <div
    className="complexlineElement"
    style={
      Object {
        "height": 300,
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`Graphs should render correctly 3`] = `""`;

exports[`Graphs should render correctly 4`] = `
<div
  className="gaugeContainer"
>
  <div
    className="gaugeElement"
  >
    <ResponsivePieCanvas
      colors={[Function]}
      data={
        Array [
          Object {
            "color": "#f4362c",
            "id": "danger",
            "value": 25,
          },
          Object {
            "color": "#ff9a09",
            "id": "warn",
            "value": 25,
          },
          Object {
            "color": "#00B35D",
            "id": "ok",
            "value": 50,
          },
        ]
      }
      enableRadialLabels={false}
      enableSlicesLabels={false}
      endAngle={-90}
      height={200}
      innerRadius={0.95}
      isInteractive={false}
      pixelRatio={2}
      sortByValue={false}
      startAngle={90}
      width={300}
    />
  </div>
  <div
    className="gaugeElement guagePercent"
  >
    <ResponsivePieCanvas
      colors={[Function]}
      data={
        Array [
          Object {
            "color": "#F4F3F1",
            "id": "reserve",
            "value": 100,
          },
          Object {
            "color": "#00B35D",
            "id": "used",
            "value": 0,
          },
        ]
      }
      enableRadialLabels={false}
      enableSlicesLabels={false}
      endAngle={-90}
      height={200}
      innerRadius={0.75}
      isInteractive={false}
      pixelRatio={2}
      sortByValue={false}
      startAngle={90}
      width={300}
    />
  </div>
  <div
    className="gaugeLabel"
  >
    <P>
      0
      %
    </P>
  </div>
</div>
`;

exports[`Graphs should render correctly 5`] = `
<div
  className="sparklineContainer"
>
  <div
    className="sparklineLabel"
  >
    <P>
      0
    </P>
  </div>
  <div
    className="sparklineElement"
  >
    <Sparklines
      data={Array []}
      height={50}
      limit={50}
      margin={0}
      preserveAspectRatio="none"
      width={400}
    >
      <SparklinesCurve
        color="#0CBEDB"
        style={Object {}}
      />
    </Sparklines>
  </div>
</div>
`;

exports[`Graphs should render correctly 6`] = `
<div
  className="streamContainer"
>
  <div
    className="streamNoData"
  >
    <P>
      No task data
    </P>
  </div>
  <div
    className="streamElement"
    style={
      Object {
        "height": 200,
      }
    }
  >
    <ResponsiveStream
      animate={false}
      axisBottom={null}
      colors={[Function]}
      data={Array []}
      enableGridX={false}
      fillOpacity={0.85}
      keys={
        Array [
          "Failed Tasks",
          "Successful Tasks",
          "Running Tasks",
          "Queued Tasks",
        ]
      }
      legends={
        Array [
          Object {
            "anchor": "top-right",
            "direction": "column",
            "itemDirection": "left-to-right",
            "itemHeight": 20,
            "itemOpacity": 0.75,
            "itemWidth": 50,
            "itemsSpacing": 0,
            "justify": false,
            "symbolShape": "square",
            "symbolSize": 5,
            "translateX": 65,
            "translateY": 0,
          },
        ]
      }
      margin={
        Object {
          "bottom": 30,
          "left": 40,
          "right": 150,
          "top": 0,
        }
      }
      offsetType="silhouette"
    />
  </div>
</div>
`;

exports[`Graphs should render correctly 7`] = `""`;
